File: Go/exercicio6.go
package main

import "fmt"

type Motor struct {
    Potencia int
}

type Carro struct {
    Marca  string
    Modelo string
    Motor  Motor
}

func main() {
    motor1 := Motor{Potencia: 150}
    carro := Carro{Marca: "Toyota", Modelo: "Corolla", Motor: motor1}
    fmt.Printf("%s %s com motor de %d HP\n", carro.Marca, carro.Modelo, carro.Motor.Potencia)
}

File: Java/exercicio6.java
class Motor {
    int potencia;

    Motor(int potencia) {
        this.potencia = potencia;
    }
}

class Carro {
    String marca;
    String modelo;
    Motor motor;

    Carro(String marca, String modelo, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.motor = motor;
    }
}

public class Main {
    public static void main(String[] args) {
        Motor motor1 = new Motor(150);
        Carro carro = new Carro("Toyota", "Corolla", motor1);
        System.out.println(carro.marca + " " + carro.modelo + " com motor de " + carro.motor.potencia + " HP");
    }
}
